src/main/java/org/apache/flink/benchmark/SerializationFrameworkBenchmarks.java

```diff
@@ -0,0 +1,451 @@
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testPojoSerializer() throws Exception {
+		LocalStreamEnvironment env =
+				StreamExecutionEnvironment.createLocalEnvironment(4);
+
+		env.addSource(new PojoSource(RECORDS_PER_INVOCATION, 10))
+				.rebalance()
+				.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testPojoSerializerRegistered() throws Exception {
+		LocalStreamEnvironment env =
+				StreamExecutionEnvironment.createLocalEnvironment(4);
+		ExecutionConfig executionConfig = env.getConfig();
+		executionConfig.registerPojoType(MyPojo.class);
+		executionConfig.registerPojoType(MyOperation.class);
+
+		env.addSource(new PojoSource(RECORDS_PER_INVOCATION, 10))
+				.rebalance()
+				.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testTupleSerializer() throws Exception {
+		LocalStreamEnvironment env =
+				StreamExecutionEnvironment.createLocalEnvironment(4);
+
+		env.addSource(new TupleSource(RECORDS_PER_INVOCATION, 10))
+				.rebalance()
+				.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testKryoSerializer() throws Exception {
+		LocalStreamEnvironment env =
+				StreamExecutionEnvironment.createLocalEnvironment(4);
+		env.getConfig().enableForceKryo();
+
+		env.addSource(new PojoSource(RECORDS_PER_INVOCATION, 10))
+				.rebalance()
+				.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testKryoSerializerRegistered() throws Exception {
+		LocalStreamEnvironment env =
+				StreamExecutionEnvironment.createLocalEnvironment(4);
+		ExecutionConfig executionConfig = env.getConfig();
+		executionConfig.enableForceKryo();
+		executionConfig.registerKryoType(MyPojo.class);
+		executionConfig.registerKryoType(MyOperation.class);
+
+		env.addSource(new PojoSource(RECORDS_PER_INVOCATION, 10))
+				.rebalance()
+				.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testAvroForcedSerializer() throws Exception {
+		LocalStreamEnvironment env =
+				StreamExecutionEnvironment.createLocalEnvironment(4);
+		env.getConfig().enableForceAvro();
+
+		env.addSource(new PojoSource(RECORDS_PER_INVOCATION, 10))
+				.rebalance()
+				.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	@OperationsPerInvocation(value = SerializationFrameworkBenchmarks.RECORDS_PER_INVOCATION)
+	public void testAvroSchemaSerializer() throws Exception {
```

Comment:
please drop the `test` prefix from the methods. It's redundant with `@Benchmark` annotation and longer names are more problematic for `codespeed` ui.